### PR TITLE
RC 3.3: merge codeql-ruby repository into github/codeql

### PR DESCRIPTION
**Pull request by aibaars**
@dbartol This pull request replays the steps I took to merge codeql-ruby into this repository. In this case I used the rc/3.3 branch of codeql and the codeql-ruby SHA of the 2.6.3 CLI release as starting points.

Below is the list of commits I made; 7741a72cc5 is the actual merge, and the rest are small tweaks to make CI work smoothly etc. The changes are best reviewed on a per-commit basis.

b79f8f1890 (HEAD -> codeql-ruby-3.3) Fix CI jobs
8cd86ae8f5 Move queries.xml to `src`
b23b3c33f6 Add a queries.xml file (for CWE coverage) docs
de38570424 Merge identical-files.json
1bf4542c89 Remove github/codeql submodule
ddbba403f8 Update CodeSpaces configuration
aeb9ace694 Add ruby to CODEOWNERS
7741a72cc5 Merge remote-tracking branch 'codeql-ruby/rc/3.3' into codeql/rc/3.3
8ce7b287d1 Update dependabot config
3554e8d105 Drop LICENSE and CODE_OF_CONDUCT.md
2de757335f Update Ruby workflows
068beeff56 Move create-extractor-pack Action
d2ea732539 Remove CodeSpaces configuration
ba32c54038 Move files to ruby subfolder
**tip of codeql rc/3.3** 702c647556 (origin/rc/3.3) Merge pull request #6904 from shati-patel/ruby-query-help
**tip of codeql-ruby** [1d58f8cd50](https://github.com/github/codeql-ruby/commit/1d58f8cd50023d7851b8ec52520640079a62ef02) (tag: codeql-cli/v2.6.3) Merge pull request #320 from github/rasmuswl/fix-hasLocationInfo-url

### CODE REVIEW

File: .github/workflows/qhelp-pr-preview.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: Query help preview
```
Comment (dbartol) [inline .github/workflows/qhelp-pr-preview.yml:1]: Don't we already have a job that does this for all languages?

File: .github/workflows/sync-files.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+name: Check synchronized files
```
Comment (dbartol) [inline .github/workflows/sync-files.yml:1]: Doesn't a workflow like this already exist?

**Review by dbartol — COMMENTED** [review]
Only a couple questions around new workflows that I thought we already had elsewhere.

File: .github/workflows/sync-files.yml (hunk 2 above)
Comment (aibaars) [reply .github/workflows/sync-files.yml:1]: Not in the codeql repository. There is likely an indirect check in the closed source CLI repository.

File: .github/workflows/qhelp-pr-preview.yml (hunk 1 above)
Comment (aibaars) [reply .github/workflows/qhelp-pr-preview.yml:1]: There is no workflow for that in this repository. There is another indirect check that verifies qhelp files, however, the logs of that job are not available to open source developers. In addition this job posts the markdown previews as a comment in the pull request for easy reviewing.

**Comment by aibaars** [comment]
> Only a couple questions around new workflows that I thought we already had elsewhere.

That's true. I don't think that is a problem for this pull request though.

File: .github/workflows/sync-files.yml (hunk 2 above)
Comment (dbartol) [reply .github/workflows/sync-files.yml:1]: This probably isn't the right branch to be adding new workflows, but it seems both harmless and useful, so let's keep it.

**Review by dbartol — APPROVED** [review]
I'm OK with the two new workflows, and everything else looks fine.

**Pull request merged**